### ISSUE TITLE
Wrong format in check_email template?

### ISSUE THREAD

**Issue by lukepass** [SymfonyCasts/reset-password-bundle#142]
Hello, this is the current check_email template:

```
{% extends 'base.html.twig' %}

{% block title %}Password Reset Email Sent{% endblock %}

{% block body %}
    <p>An email has been sent that contains a link that you can click to reset your password. This link will expire in {{ tokenLifetime|date('g') }} hour(s).</p>
    <p>If you don't receive an email please check your spam folder or <a href="{{ path('app_forgot_password_request') }}">try again</a>.</p>
{% endblock %}
```

This prints the message:

`An email has been sent that contains a link that you can click to reset your password. This link will expire in 2 hour(s).`

Which is wrong because in the email I get "1 hour".

The problem is that `tokenLifetime` contains "3600" and it's not a timestamp (thus cannot be converted to date?).

Thanks.

**Comment by jrushlow** [comment]
Good Morning @lukepass, we've had a long standing issue in this bundle with the time shown in the template. That has been fixed with the [v1.2.2](https://github.com/SymfonyCasts/reset-password-bundle/releases/tag/v1.2.2) release of `ResetPasswordBundle`. At the same time we've updated the templates for both `ResetPasswordBundle` & `VerifyEmailBundle` in Symfony's `MakerBundle` to utilize translations and fix the incorrect expiration time.

To fix this issue in existing projects:

1) Run `composer update` to upgrade to the latest versions of `ResetPasswordBundle` & `MakerBundle`
2) Manually update the following files that were generated by `make:reset-password`

```diff
// ResetPasswordController::class

    private function processSendingPasswordResetEmail(string $emailFormData, MailerInterface $mailer): RedirectResponse
    {
    .....
          
        $email = (new TemplatedEmail())
            ->from(new Address('<?= $from_email ?>', '<?= $from_email_name ?>'))
            ->to($user-><?= $email_getter ?>())
            ->subject('Your password reset request')
            ->htmlTemplate('reset_password/email.html.twig')
            ->context([
                'resetToken' => $resetToken,
-               'tokenLifetime' => $this->resetPasswordHelper->getTokenLifetime(),
            ])
        ;
```

```diff
// src/templates/reset-password/email.twig.html


-    <p>This link will expire in {{ tokenLifetime|date('g') }} hour(s).</p>
+    <p>This link will expire in {{ resetToken.expirationMessageKey|trans(resetToken.expirationMessageData, 'ResetPasswordBundle') }}.</p>

```

You should then be all set! If you run into any trouble, let us know if you have any issues and thanks for using ResetPasswordBundle!

**Comment by lukepass** [comment]
Hello and thanks for the quick answer!

Unfortunately in my checkEmail action I don't have access to `resetToken`:

    public function checkEmail(): Response
    {
        // We prevent users from directly accessing this page
        if (!$this->canCheckEmail()) {
            return $this->redirectToRoute('app_forgot_password_request');
        }

        return $this->render('reset_password/check_email.html.twig', [
            'tokenLifetime' => $this->resetPasswordHelper->getTokenLifetime(),
        ]);
    }

**Comment by jrushlow** [comment]
Ahhh - I missed the link that is displayed in the `/check-email` route. The fix above is for the email itself. I'll get an update pushed out in the next couple hours. Stay tuned!

**Comment by jrushlow** [comment]
To fix this issue in existing projects:

1) Run `composer update` to upgrade to the latest versions of `ResetPasswordBundle` & `MakerBundle`
2) Manually update the following files that were generated by `make:reset-password`

```diff
// ResetPasswordController::class

-   public function checkEmail(): Response
+   public function checkEmail(ResetPasswordToken $resetToken = null): Response
    {
        // We prevent users from directly accessing this page
-     if (!$this->canCheckEmail()) {
+     if (null === $resetToken || !$this->canCheckEmail()) {
            return $this->redirectToRoute('app_forgot_password_request');
        }

        return $this->render('reset_password/check_email.html.twig', [
-            'tokenLifetime' => $this->resetPasswordHelper->getTokenLifetime(),
+            'resetToken' => $resetToken,
        ]);
    }

    private function processSendingPasswordResetEmail(string $emailFormData, MailerInterface $mailer): RedirectResponse
    {
    .....
          
        $email = (new TemplatedEmail())
            ->from(new Address('<?= $from_email ?>', '<?= $from_email_name ?>'))
            ->to($user-><?= $email_getter ?>())
            ->subject('Your password reset request')
            ->htmlTemplate('reset_password/email.html.twig')
            ->context([
                'resetToken' => $resetToken,
-               'tokenLifetime' => $this->resetPasswordHelper->getTokenLifetime(),
            ])
        ;

-    return $this->redirectToRoute('app_check_email');
+    return $this->redirectToRoute('app_check_email', [$resetToken]);
    }
```

```diff
// src/templates/reset-password/email.twig.html


-    <p>This link will expire in {{ tokenLifetime|date('g') }} hour(s).</p>
+    <p>This link will expire in {{ resetToken.expirationMessageKey|trans(resetToken.expirationMessageData, 'ResetPasswordBundle') }}.</p>

```

```diff
// src/templates/reset-password/check_email.html.twig

-    <p>An email has been sent that contains a link that you can click to reset your password. This link will expire in {{ tokenLifetime|date('g') }} hour(s).</p>
+    <p>
+       An email has been sent that contains a link that you can click to reset your password.
+       This link will expire in {{ resetToken.expirationMessageKey|trans(resetToken.expirationMessageData, 'ResetPasswordBundle') }}.
+    </p>
    <p>If you don't receive an email please check your spam folder or <a href="{{ path('app_forgot_password_request') }}">try again</a>.</p>

```

You should then be all set! If you run into any trouble, let us know if you have any issues and thanks for using ResetPasswordBundle!

**Comment by jrushlow** [comment]
@lukepass I've refactored the example above to reflect the changes that will be coming in the next `MakerBundle` release. In the meantime, the above should take care of both templates. Thanks for bringing this to our attention!

**Comment by lukepass** [comment]
I think that there are other cases where you would be redirected to `app_check_email` without a token, for example:

        // Do not reveal whether a user account was found or not.
        if (!$user) {
            return $this->redirectToRoute('app_check_email');
        }

This wouldn't work (you are not passing the token) and it would redirect back to registration.

Moreover passing parameters with `redirectToRoute` doesn't seem to work and `$resetToken` is always null in the `checkEmail` action.

Thanks again.

**Comment by jrushlow** [comment]
@lukepass we just released v1.3.0 which adds getter/setters in the `ResetPasswordControllerTrait::class` to persist the `ResetPasswordToken` object into tthe session. I've also refactored the example above to use the new methods. This should take care of the issue.

Checkout https://github.com/SymfonyCasts/reset-password-bundle/releases/tag/v1.3.0 for full release details! Template updates in MakerBundle are soon to follow.
